File: ios/source/model/Waveform.swift
import Foundation

struct WaveformData: Codable {
    var sampleRate: Int
    var peaks: [WaveformGroup]
}

struct WaveformGroup: Codable {
    var properties: WaveformProperties
    var values: [Float]
}

struct WaveformProperties: Codable {
    var length: Int
    var algorithm: WaveformAlgorithm
    var channel: Int
}

enum WaveformAlgorithm: String, Codable {
    case min
    case max
    case rms
}

typealias Waveforms = [Id: WaveformData]

File: editor/src/model/waveform.ts
export interface WaveformPeaks {
  properties: {
    algorithm: string;
    channel: number;
    length: number;
  };
  values: number[];
}

export interface WaveformData {
  sampleRate: number;
  peaks: WaveformPeaks[];
}

export interface Waveform {
  sampleId: string;
  waveformData: WaveformData;
}
